Refactor auth

maahsnd opened the pull request:
Addresses issue #19 

Factor sign up and log in out into independent components, remove auth form component.
Redirect authenticated users to home when they attempt to access sign up / log in.

maahsnd committed:
Factor out sign up and log in

maahsnd committed:
Redir authenticated users

DONTSTOPLOVINGMEBABY commented on src/components/signupForm/SignUpForm.tsx:
```diff
@@ -54,61 +60,107 @@ function SignupForm({ onSubmit }: SignupFormProps) {
+          'Content-Type': 'application/json'
+        },
+        credentials: 'include',
+        body: JSON.stringify({
```
**Backend is currently unable to read username because username field in JSON payload is incorrectly labeled**

Change 
```typescript
body: JSON.stringify({
  // ... rest of payload
  inputUsername,
})
```
to 
```typescript
body: JSON.stringify({
    // rest of payload
  username: inputUsername, 
})
```

You could also just change inputUsername variable to have the name username, which is likely the cleaner fix

maahsnd replied:
That's what I get for find and replace all username with inputUsername... Had to differentiate from authContext username.

DONTSTOPLOVINGMEBABY reviewed (changes requested):
Backend is currently unable to read username argument from request body in signup. See snippet

maahsnd committed:
Name fields correctly
Find and replace all was applied too liberally
This was intially done to avoid collision
between authContext username and input username.
This commit returns json payload to its proper
shape, and does the same for auth context.

DONTSTOPLOVINGMEBABY commented on src/components/loginForm/LoginForm.tsx:
```diff
@@ -1,55 +1,106 @@
-  const handleSubmit = (e: React.FormEvent) => {
-    e.preventDefault();
-    onSubmit(username, password);
+  const handleSubmit = async () => {
```
`handleSubmit` needs to prevent the default behavior of the html form

Add *e* as an argument to handleSubmit with type React.FormEvent\<HTMLFormElement\> 

```typescript
  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => // ... 
```

DONTSTOPLOVINGMEBABY commented on src/components/loginForm/LoginForm.tsx:
```diff
@@ -1,55 +1,106 @@
-    e.preventDefault();
-    onSubmit(username, password);
+  const handleSubmit = async () => {
+    try {
```
Add preventDefault

```typescript
try {
  e.preventDefault()
  const response = await fetch(APIBASE + 'auth/login', { ... }) 
  // ...
} 
```

DONTSTOPLOVINGMEBABY commented on src/components/signupForm/SignUpForm.tsx:
```diff
@@ -42,7 +48,7 @@ function SignupForm({ onSubmit }: SignupFormProps) {
   };
 
-  const handleSubmit = (e: React.FormEvent) => {
+  const handleSubmit = async (e: React.FormEvent) => {
```
handleSubmit for signup looks good. Good to type it the same way as in login + its narrower type

```typescript
const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => // ... 
```

DONTSTOPLOVINGMEBABY reviewed (approved):
Fix HTML Form's default behavior when submitting in Login

DONTSTOPLOVINGMEBABY reviewed (changes requested):
HTML Form's default behavior not being prevented in Signup during submission

DONTSTOPLOVINGMEBABY commented:
Meant to request changes not approve... apologies if confusing

maahsnd committed:
Correct issues

DONTSTOPLOVINGMEBABY reviewed (approved):
Looks good!

DONTSTOPLOVINGMEBABY commented:
You should be able to merge this into main and then merge main into the branch that you're also working on that is currently working on, if that seems like what you would want to do.
